tutorials/snapcraft/snap-a-nodejs-app.md

```diff
@@ -0,0 +1,666 @@
+
+
+positive
+: It’s just a matter of preference where you want to define this `daemon` entry. We personally advise to let it under the command entry and before **plugs** to let the snap author remember that this entry is a service.
+
+
+The `simple` attribute means that it’s a classical application. The process keeps running as long as the service itself runs.
+There are variations to this, like `forking`, which is the traditional double fork way for a service to be orphaned, and thus attached to PID 1. This isn’t the recommended behavior on a modern Linux system. Nowadays, **systemd** (which is what is used by snap technology) knows how to track them efficiently, you just need to specify that you have this type of daemons.
+`oneshot`, finally, expects that the command configured will exit once it's done (won't be a long-lasting process). The service is still considered as running though.
+For those used to systemd service units, indeed, there is an one to one mapping here, handy! :)
+
+
+positive
+: `stop-command` and `timeout-command` are other daemon-related options you can set. Those definitions are available in the snapcraft syntax guide. You will get a link as a reference (last step of this tutorial).
+
+### Keeping it always running
+
+Last detail to figure out. The default behavior for a service is to restart if the applications exited in a failure state (with return code **different from 0**).
+Here, we want to always have the service running, especially if we implement a “restart” button and thus, the service exits successfully (0). By defining the following `restart-condition: always`, we achieve this desired behavior:
+
+
+```bash
+apps:
+  node-service:
+    command: bin/chuck-norris-app
+    daemon: simple
+    restart-condition: always
+    plugs: [network-bind]
+```
+
+### Running and testing
+
+Let’s give it a shot:
+
+
+```bash
+$ snapcraft prime
+$ snap try prime/
+chuck-norris-webserver 1.0.0 mounted from […]/prime
+```
+
+Head over to [http://localhost] and you should have your successful snap running your application as a service! Installing the snap started thus the service.
+
+### Where did the logs go?
+
+Services are handled by systemd. You can access to all systemd commands to handle them. However, for traditional Linux reasons, we can also follow them in `/var/log/syslog`:
+
+
+```bash
+Nov 15 14:40:11 tidus snap[14410]: Server listening on: http://localhost:80
+Nov 15 14:47:35 tidus snap[14410]: Quoted http://api.chucknorris.io/jokes/ijpnw0birv-ogszti_wuag
+```
+
+However, this can be hard to decipher: nothing tells that the log coming from this particular service, you only know it’s coming from this snap because you wrote this app! You will see as well a lot of other logs on the system. We need to know how to filter this a little bit.
+
+The service name (where you can find the service file generated in `/etc/systemd/system`) is named the following: “**snap.<snap_name>.<app_name>.service**”
```

Comment:
“**snap.<snap_name>.<app_name>.service**” renders as "**snap**...service". Should probably be changed to `snap.<snap_name>.<app_name>.service` to work fine.